src/hw_isolation_event/event.cpp

```diff
@@ -38,9 +47,71 @@ Event::Event(sdbusplus::bus::bus& bus, const std::string& objPath,
     // of this event
     associations(associationDef);
 
-    // Emit the signal for the event object creation since it deferred in
-    // interface constructor
-    this->emit_object_added();
+    if (reqDeserialize)
+    {
+        deserialize();
+    }
+    else
+    {
+        // Emit the signal for the event object creation since it deferred
+        // in interface constructor
+        this->emit_object_added();
+
+        serialize();
+    }
+}
+
+Event::~Event()
+{
+    fs::path path{fmt::format(HW_ISOLATION_EVENT_PERSIST_PATH, _eventId)};
+    if (fs::exists(path))
+    {
+        fs::remove(path);
+    }
+}
+
+void Event::serialize()
+{
+    fs::path path{fmt::format(HW_ISOLATION_EVENT_PERSIST_PATH, _eventId)};
+    try
+    {
+        std::ofstream os(path.c_str(), std::ios::binary);
+        cereal::BinaryOutputArchive oarchive(os);
+        oarchive(*this);
+    }
+    catch (const cereal::Exception& e)
+    {
+        log<level::ERR>(fmt::format("Exception: [{}] during serialize the "
+                                    "hardware isolation status event into {}",
+                                    e.what(), path.string())
+                            .c_str());
+        fs::remove(path);
+    }
+}
+
+bool Event::deserialize()
+{
+    fs::path path{fmt::format(HW_ISOLATION_EVENT_PERSIST_PATH, _eventId)};
+    try
+    {
+        if (fs::exists(path))
+        {
+            std::ifstream is(path.c_str(), std::ios::in | std::ios::binary);
+            cereal::BinaryInputArchive iarchive(is);
+            iarchive(*this);
+            return true;
+        }
+        return false;
+    }
+    catch (const cereal::Exception& e)
+    {
+        log<level::ERR>(fmt::format("Exception: [{}] during deserialize the "
+                                    "hardware isolation status event from {}",
+                                    e.what(), path.string())
+                            .c_str());
+        fs::remove(path);
+        return false;
```

Comment:
we are not doing anything with this return value

Reply:
Ack